docs/design.md

```diff
@@ -1,3 +1,99 @@
+
+If user A shares a list with user B, can user B delete that list?
+* No. Unless B is an owner, as decided by user A.
+
+If user A shares a list with user B, does user B need to accept that sharing, or will a shared list just show up?
+* Shared list just shows up - with the caveat that users have the ability to remove themselves as viewers of a shared list.
+
+Should shared lists be distinguishable in the UI from unshared lists?
+* Yes. This is reflected in the list manager
+
+If user A shares a list with user B, can user B see that list if they don't have an authenticated email address?
+
+* No. We force everyone to authenticate to see any lists (theirs or shared!)
+
+
+### User Testing + Design Iteration:
+
+We then drafted some mockups for our signup/signin pages. 
+
+At first, we thought it would be best to give users the ability to sign up or sign in on the same page. The mockup of this idea was the following:
+
+![NoTasks](./images/screenshots/signupdraft.jpg)
+
+However, upon asking our trusty suitemates as user testers, we found that the page was a little too packed. In addition, it didn't really make a lot of sense to give users a box to sign-up if all they want to do was sign-in. 
+
+Instead, we found the following flow to work better. First, users will be prompted to choose either "sign in" or "sign up." Then, depending on which they choose, they will either be prompted to create an account, or simply login with their credentials. In the case that a user mistakenly clicks sign-in instead of sign-up (or vice versa), we also allow users to toggle between screens.
+
+The mockup for the flow we decided on is the following: 
+
+![NoTasks](./images/screenshots/signupflow.jpg)
+
+We then spent a while implementing how users created and interacted with shared lists. We found via a few final user tests (with roommates) that it made most sense to denote a separate button for sharing and managing lists. That is, each user can keep track of their own lists (create, rename, edit) with one button, and manage sharing preferences with another.
+
+New Homepage:
+
+![NoTasks](./images/screenshots/newhp.jpg)
+
+We found that it became too packed to try and cram in both sharing and list editing features into one selection, so splitting it into two buttons increased ease of use for both.
+
+
+### Challenges We Faced:
+
+We had a difficult time working through the Firestore rules. At first, we got stuck on a screen telling us we didn't have permission to view any lists, including lists we knew we created.
+
+We also weren't entirely sure whether security rules were better decided at the firestore level, or if they could be done from the frontend (for instance, not giving users the ability to view lists they didn't own / shared on versus embedding a rule in Firestore). Ultimately, we found it to be more secure to write into the rules, so we decided on that. 
```

Comment:
Definitely rules because anybody could write their own front-end (with no security rules) to access your Firestore database.